src/ASM/ASMs2DLag.C

```diff
@@ -765,3 +765,20 @@ void ASMs2DLag::generateThreadGroups (const Integrand&, bool, bool)
 {
   threadGroups.calcGroups((nx-1)/(p1-1),(ny-1)/(p2-1),1);
 }
+
+
+bool ASMs2DLag::write(std::ostream& os, int) const
+{
+  os << "# LAGRANGIAN nodes=" << this->getNoNodes()
+      << " elements=" << this->getNoElms()
+      << " type=quad\n";
+  for (const Vec3& node : myCoord)
+    os << node << "\n";
+  for (const IntVec& mnpc : MNPC) {
+    for (const int& i : mnpc)
+      os << i << " ";
```

Comment:
Cosmetic, but this will add a trailing space at each line of output, suggest rewrite:
```
  for (size_t i = 0; i < mnpc.size(); i++)
    os << mnpc[i] << (i+1 == mnpc.size() ? '\n' : ' ');
```
and remove L780

Reply:
done.